contracts/contracts/core/ProviderRegistry.sol

```diff
@@ -297,15 +302,19 @@ contract ProviderRegistry is
         emit FundsDeposited(provider, msg.value);
     }
 
-    function _registerAndStake(address provider, bytes calldata blsPublicKey) internal {
+    function _registerAndStake(address provider, bytes[] calldata blsPublicKeys) internal {
         require(!providerRegistered[provider], ProviderAlreadyRegistered(provider));
         require(msg.value >= minStake, InsufficientStake(msg.value, minStake));
-        require(blsPublicKey.length == 48, InvalidBLSPublicKeyLength(blsPublicKey.length, 48));
-        
-        eoaToBlsPubkey[provider] = blsPublicKey;
+        require(blsPublicKeys.length > 0, AtLeastOneBLSKeyRequired());
+        uint256 numKeys = blsPublicKeys.length;
+        for (uint256 i = 0; i < numKeys; ++i) {
```

Comment:
`bytes blsPublicKey = blsPublicKeys[i];`
Then use this variable. That way you will have less storage reads